web/satellite/src/api/buckets.ts

```diff
@@ -50,10 +50,25 @@ export class BucketsApiGql extends BaseGql implements BucketsApi {
 
         const response = await this.query(query, variables);
 
-        return this.fromJson(response.data.project.bucketUsages);
+        return this.getBucketPage(response.data.project.bucketUsages);
     }
 
-    private fromJson(bucketPage): BucketPage {
-        return new BucketPage(bucketPage.bucketUsages, bucketPage.search, bucketPage.limit, bucketPage.offset, bucketPage.pageCount, bucketPage.currentPage, bucketPage.totalCount);
+    private getBucketPage(page: any): BucketPage {
+        if (!page) {
+            return new BucketPage();
+        }
+
+        const bucketPage: BucketPage = new BucketPage();
```

Comment:
const buckets = page.bucketUsages.map(key => new Bucket(key.bucketName, key.storage, key.egress, key.objectCount, key.since, key.before));

 return new BucketPage(x,page.search, page.search, ...);

Reply:
done

Reply:
done